feat(sdk): SDK Backward Compatibility support via Hosted Bundle (#60554)
metabase/metabase: docs/embedding/sdk/next-js.md

Some notes on using the Embedded analytics SDK with [Next.js](https://nextjs.org/). The SDK is tested to work with Next.js 14, although it may work with other versions.

See a [sample Next.js app that uses the SDK](https://github.com/metabase/metabase-nextjs-sdk-embedding-sample).

## SDK components with Server Side Rendering (SSR) or React Server Components

> The SDK components render only on the client side and not on the server side.

Reviewer: ```suggestion
The SDK components render only on the client side, not on the server side.
```

> The SDK components render only on the client side and not on the server side.

Reviewer: I'm okay with mentioning the version, as people might be jumping versions.

> The SDK components render only on the client side and not on the server side.

Author: @npretto just to clarify. From what i know the SDK always was rendered only on the client side, so not sure why we should mention version for this line. Though i can mention version number (57) where the Nextjs compat is no longer required

> The SDK components render only on the client side and not on the server side.

Author: ```
## SDK components with Server Side Rendering (SSR) or React Server Components

As of Embedding Analytics SDK v57, SDK components automatically skip server-side rendering (SSR) and render only on the client.

### Compatibility layer for Server Side Rendering (SSR) (DEPRECATED)

As of Embedding analytics SDK 57, the compatibility layer for server-side rendering (SSR) is deprecated and no longer required. If you use the compatibility layer, change your imports from `@metabase/embedding-sdk-react/next` to `@metabase/embedding-sdk-react`.
```

> The SDK components render only on the client side and not on the server side.

Author: https://github.com/metabase/metabase/pull/62164/commits/c8628b149483e9acce385d0f51d9bb8497304848

### Compatibility layer for server-side rendering (SSR) (DEPRECATED)

> _Note_: Compatibility layer for server-side rendering (SSR) is no more needed and was deprecated. If you use it, change your imports from `@metabase/embedding-sdk-react/next` to `@metabase/embedding-sdk-react`.

Reviewer: ```suggestion
The compatibility layer for server-side rendering (SSR) is no longer necessary and has been deprecated. If you use the compatibility layer, change your imports from `@metabase/embedding-sdk-react/next` to `@metabase/embedding-sdk-react`.
```

Author: https://github.com/metabase/metabase/pull/62164/commits/c8628b149483e9acce385d0f51d9bb8497304848
